java/clients/src/main/java/sleeper/clients/admin/deploy/UpdateExistingInstance.java

```diff
@@ -0,0 +1,125 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.clients.admin.deploy;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import software.amazon.awssdk.services.s3.S3Client;
+
+import sleeper.configuration.properties.InstanceProperties;
+import sleeper.configuration.properties.local.LoadLocalProperties;
+import sleeper.configuration.properties.local.SaveLocalProperties;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+import static sleeper.configuration.properties.UserDefinedInstanceProperty.JARS_BUCKET;
+import static sleeper.configuration.properties.UserDefinedInstanceProperty.REGION;
+import static sleeper.configuration.properties.UserDefinedInstanceProperty.VERSION;
+
+public class UpdateExistingInstance {
+    private final Path scriptsDirectory;
+    private final String instanceId;
+    private final AmazonS3 s3;
+    private final S3Client s3v2;
+
+    private UpdateExistingInstance(Builder builder) {
+        scriptsDirectory = builder.scriptsDirectory;
+        instanceId = builder.instanceId;
+        s3 = builder.s3;
+        s3v2 = builder.s3v2;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static void main(String[] args) throws IOException, InterruptedException {
+        if (2 != args.length) {
+            throw new IllegalArgumentException("Usage: <scripts-dir> <instance-id>");
+        }
+        Path scriptsDirectory = Path.of(args[0]);
+
+        AmazonS3 s3 = AmazonS3Client.builder().build();
+        try (S3Client s3v2 = S3Client.create()) {
+            builder().s3(s3)
+                    .s3v2(s3v2)
+                    .instanceId(args[1])
+                    .scriptsDirectory(scriptsDirectory)
+                    .build().update();
+        }
+    }
+
+    public void update() throws IOException, InterruptedException {
+        // Get instance properties from s3
+        Path generatedDirectory = scriptsDirectory.resolve("generated");
+        Path jarsDirectory = scriptsDirectory.resolve("jars");
+        SaveLocalProperties.saveFromS3(s3, instanceId, generatedDirectory);
+        InstanceProperties properties = LoadLocalProperties.loadInstanceProperties(new InstanceProperties(), generatedDirectory);
+
+        // Upload JARs to bucket
+        SyncJars syncJars = SyncJars.builder()
+                .s3(s3v2)
+                .jarsDirectory(jarsDirectory)
+                .region(properties.get(REGION))
+                .bucketName(properties.get(JARS_BUCKET))
+                .build();
+        syncJars.sync();
```

Comment:
This will remove the old jars, as well as any user jars. This needs to only remove old jars after the deployment finishes, and leave user jars there.